src/core/api/ip6_api.cpp

```diff
@@ -288,3 +269,45 @@ otError otIp6SelectSourceAddress(otInstance *aInstance, otMessageInfo *aMessageI
 exit:
     return error;
 }
+
+#if OPENTHREAD_CONFIG_ENABLE_SLAAC
+
+otError otIp6SetSlaacEnabled(otInstance *aInstance, bool aEnabled)
+{
+    Instance &    instance = *static_cast<Instance *>(aInstance);
+    Utils::Slaac &slaac    = instance.Get<Utils::Slaac>();
+
+    if (aEnabled)
+    {
+        slaac.Enable();
+    }
+    else
+    {
+        slaac.Disable();
+    }
+
+    return OT_ERROR_NONE;
+}
+
+otError otIp6SetSlaacPrefixFilter(otInstance *aInstance, otIp6SlaacPrefixFilter aFilter)
+{
+    Instance &instance = *static_cast<Instance *>(aInstance);
+
+    instance.Get<Utils::Slaac>().SetFilter(aFilter);
+
+    return OT_ERROR_NONE;
+}
+
+#else // OPENTHREAD_CONFIG_ENABLE_SLAAC
+
+otError otIp6SetSlaacEnabled(otInstance *, bool)
```

Comment:
Do we need this implementation? Without this, if OpenThread is not properly configured, user will find this error during compile time, which I think it's better to find during runtime.

Reply:
I followed the current model we use for other public APIs (e.g. `commisssioner_api`). 

But I agree and like your suggestion much better. Build-time failure is much better than run-time errors. I think we should adopt this instead (probably for other APIs as well). 

Thoughts?

Reply:
I always prefer compile-time checks wherever possible.

Note that the current model with the Commissioner API originated from #2041, where platform code was making calls to the Commissioner API. However, since the Commissioner API is conditional on `OPENTHREAD_FTD`, there was no way for the platform code to know whether or not the Commissioner API was available.

Should we simply remove the `OPENTHREAD_FTD` conditionals from the Commissioner API source?

Reply:
I see. Sounds ok to me. 
I changed this PR to use build-time error.